### PR TITLE
add new description of envs, mostly remove `rad env init azure` 

### PR DESCRIPTION
**Pull request by emily-potyraj**
- also removes `rad env init dev` for now 
- places that listed the 3 `rad env init` options got slimmed down to just `rad env init kubernetes`
- there are a few `rad env init azure` references left (e.g. in the CI/CD example), but they're off the main learning path 
  - leaving https://github.com/project-radius/radius/issues/2649 open for now for tracking those last 2-3 docs changes we have to make in v0.12
- also made a couple minor changes to prep the Quickstart section for v0.12

Especially seeking group feedback on the new env overview doc (which is based mostly on Izzy's draft for that page)
docs/content/concepts/environments-concept/_index.md

### CODE REVIEW

File: docs/content/concepts/environments-concept/_index.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+---
+
+## Introduction
+Radius environments can be thought of as a prepared landing zone for applications. Creating and configuring an environment results in a prepared pool of compute, networking, and dependency resources like databases. Then, a Radius application can be deployed into the environment, "binding" the app to the infrastructure. 
+
+Because environment definitions can be codified, central teams can define environment templates that let their dev teams hydrate fully-functioning environments in a self-service way - while following organization best practices like security confiugration. 
+
+<img src="env-template-example.png" alt="Diagram of example contents for an environment template. It contains dependencies like Dapr, recipes for infrastructure, security configuration, diagnostics and logging, and compute runtime." width="200" />
+
+## Configuring environments
```
Comment (ihood) [inline docs/content/concepts/environments-concept/_index.md:16]: I think including a goal statement right after this headline would be useful to users. Something that orients them as to what this section's purpose is. Ex: "There are three main steps to configure a Radius environment, and by completing these steps a user will have successfully initialized an environment."

File: docs/content/concepts/environments-concept/_index.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+- If needed, install a Radius control plane
+    - Install Radius control plane services via a set of Helm charts 
+- Create a new environment 
+    - Create a namespace to hold the environment resource 
+    - Create an environment resource
+    - Execute any environment installation templates (e.g. to install Dapr) 
+- Walk users through configuring cloud providers to connect to public clouds 
+
+Each AKS or EKS cluster requires 1 Radius control plane, but it may contain multiple Radius environmnents if desired. 
+
```
Comment (ihood) [inline docs/content/concepts/environments-concept/_index.md:41]: I think a statement showing what success looks like at this point would be useful to users. Or someway they could verify their environment is correctly configured (maybe just a simple 'rad env list').

File: docs/content/concepts/environments-concept/_index.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+## Introduction
+Radius environments can be thought of as a prepared landing zone for applications. Creating and configuring an environment results in a prepared pool of compute, networking, and dependency resources like databases. Then, a Radius application can be deployed into the environment, "binding" the app to the infrastructure. 
+
+Because environment definitions can be codified, central teams can define environment templates that let their dev teams hydrate fully-functioning environments in a self-service way - while following organization best practices like security confiugration. 
+
+<img src="env-template-example.png" alt="Diagram of example contents for an environment template. It contains dependencies like Dapr, recipes for infrastructure, security configuration, diagnostics and logging, and compute runtime." width="200" />
+
+## Configuring environments
+### 1. Define the environment
```
Comment (ihood) [inline docs/content/concepts/environments-concept/_index.md]: Does this step require any action from the user? If so, I think we should state what the actions are. If not, I think we should somehow state that no action is needed from the user here.

File: docs/content/getting-started/tutorial/_index.md [hunk 4]
```diff
@@ -37,37 +37,10 @@ In this tutorial, you will:
...
-
-```sh
-rad env init dev
-```
-
-{{% /codetab %}}
-
-{{% codetab %}}
-A Kubernetes envionment can run in any Kubernetes cluster. Make sure you have set the correct default kubectl context, and then run:
+A Radius Kubernetes envionment can run in a Kubernetes cluster running on any platform. Make sure you have set the correct default kubectl context, and then run:
```
Comment (ihood) [inline docs/content/getting-started/tutorial/_index.md]: Maybe would be useful to have the exact command to make sure the current context is the cluster. Thinking back to Donovan's walkthrough and how we don't want to assume users have experience with kubectl

**Review by ihood — COMMENTED** [review]
I added some suggestions (p.s. I really like the image on the environments page, I think it will help users more quickly digest exactly what an environment is!)

File: docs/content/concepts/environments-concept/_index.md (hunk 1 above)
Comment (emily-potyraj) [reply docs/content/concepts/environments-concept/_index.md:16]: good idea, done.

**Review by ihood — APPROVED** [review]
looks good!

**Pull request merged**